url.py

```diff
@@ -322,6 +322,10 @@ def unpunycode(self):
     ###########################################################################
     # Information about the domain
     ###########################################################################
+    def hostname(self):
```

Comment:
Ok, I see now, `punycode()` doesn't return the string, it returns `self`. However, this still entails a conflict with `pld()`. If someone calls `url.punycode().pld()` this will not work if the URL gets changed by punycoding. The same thing is true for `tld()`. Similarly, `url.pld()` will fail if the input was previously punycoded.